.github/actions/valgrind_run/action.yml

```diff
@@ -14,5 +14,8 @@ runs:
       run: |
         pyccel --language=c --flags="-g -O0" leaks_check.py
         valgrind --leak-check=full --error-exitcode=1 ./leaks_check
+        pyccel-clean -p
```

Comment:
The name of this job indicates that it is for `stc containers`, maybe we can add a new job for `gFTL`.

Reply:
Fixed : ded85d518

I prefer to keep similar tests grouped I think it could be clearer if we have a large number of leak files at some point.

But actually I've never worked out how to distinguish these steps inside an action in the output of a test so I don't think it's super critical